### ISSUE TITLE
`no-constant-condition` should respect partial conditions too

### ISSUE THREAD

**Issue by pvamshi** [eslint/eslint#5693]
Not sure if I got the title correctly, but we are valuating for a whole condition but if its part of multiple conditions, we are not checking them. 

I was working on #5228 and came across this bug . Please suggest if its a valid bug and I can take care fixing it. 

**What version of ESLint are you using?**
1.5.1
**What parser (default, Babel-ESLint, etc.) are you using?**
default
**Please show your full configuration:**
```js
{
  "rules": {
    "no-constant-condition": 2   // disallow use of constant expressions in conditions
  }
}
```
**What did you do? Please include the actual source code causing the issue.**
```js
if(false && abc==='str'){
}
```
**What did you expect to happen?**
Expect it to throw error
**What actually happened? Please include the actual, raw output from ESLint.**
no error

**Comment by nzakas** [comment]
So you're saying this should flag the `false` because it short circuits? If so, I agree.

**Comment by platinumazure** [comment]
I'm wondering if this should be a new rule maybe? `no-constant-condition` flags constant conditions on if statements and other clear Boolean-context syntax locations; something like `false && foo` could also be flagged on an AssignmentExpression or VariableDeclarator as well.

But then the question would become, how to make sure `no-constant-condition` also catches it...

**Comment by nzakas** [comment]
@platinumazure sorry, I can't quite follow your rationale for a new rule. The example code in the original description is a constant condition that we are missing.

**Comment by pvamshi** [comment]
@nzakas Yes , after fixing #5228 I will work on this .

**Comment by pvamshi** [comment]
Created a pull request.
